modules/assignment/src/test/java/com/intuit/wasabi/assignment/impl/AssignmentsImplTest.java

```diff
@@ -1670,6 +1670,46 @@ public void testGetSingleAssignmentNullAssignmentRapidExperimentCapReached() {
         verify(metadataCache, times(1)).refresh();
     }
 
+    @Test
+    public void testExperimentFetchSameNameButDeleted() {
+
+        //Create input objects
+        Application.Name appName = Application.Name.valueOf("Test");
+        Experiment.ID deletedExperimentId = Experiment.ID.newInstance(); //Experiment.ID.valueOf("");
+        Experiment.ID runningExperimentId = Experiment.ID.newInstance(); //Experiment.ID.valueOf("");
+        Experiment.Label commonExperimentLabel = Experiment.Label.valueOf("SameNamePresetTwice");
+
+        Experiment deletedExperiment = mock(Experiment.class, RETURNS_DEEP_STUBS);
+        when(deletedExperiment.getID()).thenReturn(deletedExperimentId);
+        when(deletedExperiment.getEndTime().getTime()).thenReturn(new Date().getTime() + 1000000L);
+        when(deletedExperiment.getLabel()).thenReturn(commonExperimentLabel);
+        when(deletedExperiment.getState()).thenReturn(Experiment.State.DELETED);
+        when(deletedExperiment.getIsRapidExperiment()).thenReturn(false);
+
+        Experiment runningExperiment = mock(Experiment.class, RETURNS_DEEP_STUBS);
+        when(runningExperiment.getID()).thenReturn(runningExperimentId);
+        when(runningExperiment.getEndTime().getTime()).thenReturn(new Date().getTime() + 1000000L);
+        when(runningExperiment.getLabel()).thenReturn(commonExperimentLabel);
+        when(runningExperiment.getState()).thenReturn(Experiment.State.RUNNING);
+        when(runningExperiment.getIsRapidExperiment()).thenReturn(false);
+
+        PrioritizedExperimentList pExpList = new PrioritizedExperimentList();
+        pExpList.addPrioritizedExperiment(PrioritizedExperiment.from(runningExperiment, 1).build());
+        Optional<PrioritizedExperimentList> prioritizedExperimentListOptional = Optional.of(pExpList);
+
+        //Mock interactions with the cache
+        when(metadataCache.getPrioritizedExperimentListMap(appName)).thenReturn(prioritizedExperimentListOptional);
+        when(metadataCache.getExperimentById(runningExperimentId)).thenReturn(Optional.of(runningExperiment));
+        when(metadataCache.getExperimentById(deletedExperimentId)).thenReturn(Optional.of(deletedExperiment));
+
+        //Make actual call
+        Experiment returnedExperiment = assignmentsImpl.getExperiment(appName, commonExperimentLabel);
+
+        //Verify expected result
+        assertNotNull(returnedExperiment);
+        assertThat(returnedExperiment.getID(), is(runningExperimentId));
+    }
```

Comment:
@abhishekjain88 definitely, but I would think that as low priority and add in backlog, as metadataCache has been enabled in PROD for long time now and it has survived a tax-season too so it will be very unlikely that we would disable metadataCache.